versions/3.0.md

```diff
@@ -1905,91 +1903,10 @@ In the case of an `operationId`, it MUST be unique and resolved in the scope of
-missingValue: null
-```
-
-In situations where variables appear in an array, an array of variables will be extracted.
-For example:
-
-```json
-[
-  { "color": "red" },
-  { "color": "green" },
-  { "color": "blue" }
-]
-```
-
-will be extracted as such:
-
-```json
-color: ["red", "green", "blue"]
-```
-
-The variables generated can be used in locations prescribed by the definition.
-
-
-##### <a name="variableSubstitution"></a>Variable Substitution
-In all cases, _variables_ from request and responses may be substituted for link generation.
-The table below provides examples of variable expressions and examples of their use in a value:
-
-Source Location | variable expression | example reference | notes
----|:---|:---|:---
-HTTP Method            | `$method`         | `/users/{$method}`                | The allowable values for the `$method` will be those for the HTTP operation 
-Requested content type | `$request.header.accept`        | `/users/3?format={$request.header.accept}`      |  
-Request parameter      | `$request.path.id`        | `/users/{$request.path.id}`            | Request parameters MUST be declared in the `parameters` section for the operation or they cannot be used in substitution.  This includes request headers
-Request body           | `$request.body`    | `/users/{$request.body#/user/uuid}` | For operations which accept payloads, references may be made to portions of the `requestBody` or the entire body itself
-Request URL            | `$url`            | `/track?url={$url}`               | 
-Response value         | `$response.body`       | `{$response.body#/uuid}`                | Only the payload in the response can be accessed with the `$response` syntax.  
-Response header        | `$response.header` | `{$response.header.Server}`        | Single header values only are available
-
-From the request, the `parameter`s used in calling the operation are made available through the `$request` syntax.
-For responses, the response payload may be used with the `$response` syntax.
-For both requests and responses, values will be substituted in the link in sections designated with a variable expression, surrounded by curly brackets `{}`. 
-
-The variable expression is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax
-
-```
-      expression = ( "$url" | "$method" | "$request." [ source ] | "$response." [ source ])
-      source = ( header-reference | query-reference | path-reference | body-reference )  
-      header-reference = "header." token
-      query-reference = "query." name  
-      path-reference = "path." name
-      body-reference = "body#" fragment
-      fragment = a JSON Pointer [RFC6901](https://tools.ietf.org/html/rfc6901)
-      name = *( char )
-      char = as per [RFC7159](https://tools.ietf.org/html/rfc7159#section-7)
-      token = as per [RFC7230](https://tools.ietf.org/html/rfc7230#section-3.2.6)
-```
-
-The `name` identifier is case-sensitive, whereas `token` is not. 
```

Comment:
I suspect the behavior of an unresolved expression should be determined based on its usage.  For links, I think that parameter values should not be provided and for callbacks the callback should be disabled.  However, I feel we need more real world experience to be able to make this call.